tests/test_annotations.py

```diff
@@ -0,0 +1,28 @@
+from typing import Any, Dict, List
+
+import pytest
+
+
+@pytest.mark.parametrize(
+    ("values", "show_only", "expected_annotations"),
+    [
+        (
+            {"serviceAccount": {"serviceAccountAnnotations": {"foo": "bar"}}},
+            ["templates/service-account.yaml"],
+            {"foo": "bar"},
+        ),
+    ],
+)
+def test_annotations(
+    chart_resources: List[Dict[str, Any]], expected_annotations: Dict[str, str]
+) -> None:
+    """Make sure that annotations are working"""
+    # based on apache/airflow:chart/tests/tests_annotations::test_annotations_are_added
+
+    # there's only one resource because we sepecified show_only.
+    assert len(chart_resources) == 1
+    service_account = chart_resources[0]
+
+    for name, value in expected_annotations.items():
+        assert name in service_account["metadata"]["annotations"]
+        assert value == service_account["metadata"]["annotations"][name]
```

Comment:
I had experience working with somewhat similar Unit and Integration Tests written for K8s in another project for the LF exams.

While it looks powerful at the first though, I don't think it's the best fit for a simple project like `stackstorm-ha`.
The disadvantage to me is spaghetti and confusion when debugging, reading these tests, maintaining them or even asking someone to create a test. The amount of indirection and complexity is excessive.


At this point, I'd better go with the simplest possible approach. Even doing helm template, `yq` or `jq` with `jsonpath` that could be ran from the CLI would be a better, when it's much easier to debug and connect the dots from Helm template itself to the Helm template test.

Yaml to Yaml is simple as that.

Reply:
Doing things from the CLI is not simpler! It's a royal pain to maintain tests in bash and I hate writing tests that way.

Reply:
So, yet again, you and I have different definitions of simple.